regcomp.c

```diff
@@ -24850,6 +24878,383 @@ S_parse_uniprop_string(pTHX_
+            found_matches = TRUE;
+        } /* End of loop through the non-algorithmic names string */
+    }
+
+    SvREFCNT_dec_NN(names_string);
+
+    /* There are also character names not in 'names_string'.  These are
+     * algorithmically generatable.  Try this pattern on each possible one.
+     * (khw originally planned to leave this out given the large number of
+     * matches attempted; but the speed turned out to be quite acceptable
+     *
+     * There are plenty of opportunities to optimize to skip many of the tests.
+     * beyond the rudimentary ones already here */
+
+    /* First see if the subpattern matches any of the algorithmic generatable
+     * Hangul syllable names.
+     *
+     * We know none of these syllable names will match if the input pattern
+     * requires more bytes than any syllable has, or if the input pattern only
+     * matches an empty name, or if the pattern has something it must match and
+     * one of the characters in that isn't in any Hangul syllable. */
+    if (    prog->minlen <= (SSize_t) syl_max_len
+        &&  prog->maxlen > 0
+        && ( ! must
+            || SvCUR(must) == 0
+            || strspn(SvPVX(must), "\n ABCDEGHIJKLMNOPRSTUWY") == SvCUR(must)))
+    {
+        /* These constants, names, values, and algorithm were taken from the
+         * Unicode standard, version 5.1, section 3.12, and should never
+         * change. */
+        const char * JamoL[] = {
+            "G", "GG", "N", "D", "DD", "R", "M", "B", "BB",
+            "S", "SS", "", "J", "JJ", "C", "K", "T", "P", "H"
+        };
+        const char * JamoV[] = {
+            "A", "AE", "YA", "YAE", "EO", "E", "YEO", "YE", "O", "WA",
+            "WAE", "OE", "YO", "U", "WEO", "WE", "WI", "YU", "EU", "YI",
+            "I"
+        };
+        const char * JamoT[] = {
+            "", "G", "GG", "GS", "N", "NJ", "NH", "D", "L",
+            "LG", "LM", "LB", "LS", "LT", "LP", "LH", "M", "B",
+            "BS", "S", "SS", "NG", "J", "C", "K", "T", "P", "H"
+        };
+
+        const int SCount = 11172;
+        const int SBase = 0xAC00;
+        const int LBase = 0x1100;
+        const int VBase = 0x1161;
+        const int TBase = 0x11A7;
+        const int VCount = 21;
+        const int TCount = 28;
+        const int NCount = VCount * TCount;
+
+        syllable_name = newSV(syl_max_len);
+        sv_setpvn(syllable_name, hangul_prefix, hangul_prefix_len);
+
+        for (i = 0; i < SCount; i++) {
+            const int L = LBase + i / NCount;
+            const int V = VBase + (i % NCount) / TCount;
+            const int T = TBase + i % TCount;
+
+            /* Truncate back to the prefix, which is unvarying */
+            SvCUR_set(syllable_name, hangul_prefix_len);
+
+            sv_catpv(syllable_name, JamoL[L - LBase]);
+            sv_catpv(syllable_name, JamoV[V - VBase]);
+            sv_catpv(syllable_name, JamoT[T - TBase]);
+
+            if (execute_wildcard(subpattern_re,
```

Comment:
This can die, which would leak anything you haven't yet refcnt_dec'd.

Reply:
I'll mortalize it